### PR TITLE
Fix zeroex BNB: Switch maker and taker token

### PR DESCRIPTION
**Pull request by aalan3**
Intention is to fix this issue https://dune.com/queries/3069651 but I'm not sure if this is correct or not

### CODE REVIEW

**Comment by Hosuke** [comment]
Let's take tx_hash 0x4870e8b93cac7cc25dbdd40d0858f6f73518b635a3e32a8e76fc4bdb74711f48 for example:
[bscscan](https://bscscan.com/tx/0x4870e8b93cac7cc25dbdd40d0858f6f73518b635a3e32a8e76fc4bdb74711f48#eventlog)

[Dune query result](https://dune.com/queries/3069635)

File: models/zeroex/bnb/zeroex_bnb_api_fills.sql [hunk 1]
```diff
@@ -302,8 +302,8 @@ uni_v2_swap as (
         , 0xdef1c0ded9bec7f1a1670819833240f027b25eff AS taker
         , z.taker_token
         , z.maker_token
-        , greatest(bytearray_to_uint256(bytearray_substring(DATA, 45, 20)), bytearray_to_uint256(bytearray_substring(DATA, 13, 20))) AS taker_token_amount_raw
-        , greatest(bytearray_to_uint256(bytearray_substring(DATA, 77, 20)), bytearray_to_uint256(bytearray_substring(DATA, 109, 20))) AS maker_token_amount_raw
+        , greatest(bytearray_to_uint256(bytearray_substring(DATA, 45, 20)), bytearray_to_uint256(bytearray_substring(DATA, 13, 20))) AS maker_token_amount_raw,
+        , greatest(bytearray_to_uint256(bytearray_substring(DATA, 77, 20)), bytearray_to_uint256(bytearray_substring(DATA, 109, 20))) AS taker_token_amount_raw ,
```
Comment (Hosuke) [inline models/zeroex/bnb/zeroex_bnb_api_fills.sql:306]:
```suggestion
        , greatest(bytearray_to_uint256(bytearray_substring(DATA, 45, 20)), bytearray_to_uint256(bytearray_substring(DATA, 13, 20))) AS maker_token_amount_raw
        , greatest(bytearray_to_uint256(bytearray_substring(DATA, 77, 20)), bytearray_to_uint256(bytearray_substring(DATA, 109, 20))) AS taker_token_amount_raw
```

**Comment by Hosuke** [comment]
If we focus on last 7 days irregular large amount trades, we can see those trades are having type `direct_uniswapv2`:
https://dune.com/queries/3069859

I think the switching in maker/taker_token_amount_raw in `uni_v2_swap` CTE is the fix for the issue.

**Comment by Hosuke** [comment]
There may be other parts need to be fixed. I am still debugging.
https://dune.com/queries/3069889

**Comment by aalan3** [comment]
For example something like [this](https://dune.com/queries/3069974?blockchain=bnb&category=decoded_project&namespace=pancakeswap_v2&contract=PancakePair&blockchains=bnb&abi=Swap)

**Comment by aalan3** [comment]
@RantumBits do you have any insight here?

**Comment by Hosuke** [comment]
I think there might be problem with `prices.usd` feed.

File: models/zeroex/bnb/zeroex_bnb_api_fills.sql [hunk 2]
```diff
@@ -317,9 +319,18 @@ uni_v2_swap as (
...
+, uni_v2_swap as (
+    SELECT *
+        ,CASE WHEN amount0Out = UINT256 '0'
+            THEN amount1Out ELSE amount0Out
+        END AS maker_token_amount_raw
+        ,CASE WHEN amount0In = UINT256 '0' OR amount1Out = UINT256 '0'
+            THEN amount1In ELSE amount0In
+        END AS taker_token_amount_raw
+    FROM uni_v2_swap_raw
+)
```
Comment (Hosuke) [inline models/zeroex/bnb/zeroex_bnb_api_fills.sql:333]: I tried a new implementation following `uniswap_v2_ethereum_trades` to see if this fix the issue.

**Comment by RantumBits** [comment]
I just noticed the issue here and am checking on this @Hosuke

**Comment by jeff-dude** [comment]
> I just noticed the issue here and am checking on this @Hosuke

@RantumBits i was about to close this and open a new PR with the fix (or at least what i believe is right). i can provide more context there and tag you?

**Comment by RantumBits** [comment]
> > I just noticed the issue here and am checking on this @Hosuke
> 
> @RantumBits i was about to close this and open a new PR with the fix (or at least what i believe is right). i can provide more context there and tag you?

yes, that's great

**Comment by Hosuke** [comment]
I think my latest version works with correct `volume_usd` and `taker/maker_token_amount_raw`.
It may not be the simplest way to implement the `direct_uniswapv2` CTE.

But you can use these codes as a reference.

**Pull request closed without merging**